clang/lib/Sema/Sema.cpp

```diff
@@ -1579,7 +1579,7 @@ NamedDecl *Sema::getCurFunctionOrMethodDecl() const {
 }
 
 LangAS Sema::getDefaultCXXMethodAddrSpace() const {
-  if (getLangOpts().OpenCL)
+  if (getLangOpts().OpenCL || getLangOpts().OpenCLGenericAddressSpace)
```

Comment:
I think this whole thing is just working around a defect in how OpenCL was implemented. There should be no difference between generic and default address space

Reply:
I think it's because the DefaultAS applies to every single language and it wasn't deemed proper to modify C/C++ behavior? I'm not an expert, maybe if @AnastasiaStulova chimes in. Regardless, I'm mostly just working with the smallest number of changes with what we've got that gives me the desired result.